### PR TITLE
Allow updating email via SCIM

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Allow updating emails for users synced via SCIM. Some clients' AD integrations are having issues with some synchronisation caused by email changes.

Related PR:
- https://github.com/Budibase/budibase-pro/pull/304

## Addresses
- [BUDI-8416 - Allow updating email via SCIM
](https://linear.app/budibase/issue/BUDI-8416/allow-updating-email-via-scim)

## Launchcontrol
Allow updating emails for users synced via SCIM.

### CODE REVIEW

File: packages/backend-core/src/users/test/db.spec.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+              _rev: expect.stringMatching(/^2-\w+/),
+            })
+          )
+        })
+      })
+
+      it("updating emails frees previous emails", async () => {
+        await config.doInTenant(async () => {
+          const previousEmail = user.email
+          const newEmail = generator.email({})
```
Comment (mike12345567) [inline packages/backend-core/src/users/test/db.spec.ts:160]: why the empty object to the generator?

**Review by mike12345567 — APPROVED** [review]
LGTM

File: packages/backend-core/src/users/test/db.spec.ts (hunk 1 above)
Comment (adrinr) [reply packages/backend-core/src/users/test/db.spec.ts:160]: We added a custom lint rule that "requires" a parameter. An empty object works, for this

Comment (samwho) [reply packages/backend-core/src/users/test/db.spec.ts:160]: 😭

**Pull request merged**